php/ext/google/protobuf/def.c

```diff
@@ -215,30 +234,215 @@ static void descriptor_init_c_instance(Descriptor *desc TSRMLS_DC) {
   desc->json_serialize_handlers_preserve = NULL;
 }
 
+PHP_METHOD(Descriptor, getFullName) {
+  Descriptor *intern = UNBOX(Descriptor, getThis());
+  const char* fullname = upb_msgdef_fullname(intern->msgdef);
+
+  // Prepend . to fullname.
```

Comment:
Do you think it would be better to strip this in the PHP implementation, instead of adding it in the c implementation? I'm not sure what the tradeoffs are here, but if there are no other reasons then it seems right to try to optimize the c implementation.